# feat: ajout du chatbot Fellah-GPT (RAG) pour l’agriculture marocaine

achrafbouiguigne opened the pull request:
# feat: ajout du chatbot Fellah-GPT (RAG) pour l’agriculture marocaine

Dans le cadre du test technique, j’ai choisi de contribuer en intégrant une fonctionnalité IA en lien avec l’agriculture marocaine.

## 🎯 Objectif

Développer un chatbot intelligent, **Fellah-GPT**, basé sur une architecture **RAG (Retrieval-Augmented Generation)**, capable de répondre à des questions sur l’agriculture marocaine à partir de documents PDF locaux.

---

## 🛠️ Changements principaux

### 🔙 Backend

- 📁 Nouveau dossier `RAG/` contenant :
  - `rag.py` : pipeline RAG avec **LangChain**, **FAISS**, **Google Generative AI**
  - 3 documents PDF : `agri_laws.pdf`, `agri_climate.pdf`, `corps_technique.pdf`
  - `.env` dédié à la clé API
  - `requirements.txt` spécifique au module RAG

- 🔗 Ajout de la route `/rag` dans `app.py`

---

### 🖥️ Frontend

- 📂 `rag.js` : fichier d’appel à l’API `/rag`
- 💬 Composant React `RAGchat.jsx` avec un bouton flottant
- 🧩 Intégration dans `Home.jsx`
- ➕ Dépendance ajoutée : `lucide-react`

---

## 🧪 Instructions pour tester

### Backend (RAG)
```bash
cd backend/RAG
pip install -r requirements.txt
````
````
Créer un fichier .env :

env
````bash


GOOGLE_API_KEY=AIzaSyCAi1XPo_dBVxTjTln7LVJVMvBgRd1Qzgk
GOOGLE_MODEL_URL=https://generativelanguage.googleapis.com/v1beta/models/gemini-2.0-flash`
````


Lancer le moteur RAG :
````bash
python rag.py
````
###Front end 
````bash
cd frontend
npm install lucide-react
npm run dev
````
💬 **Fonctionnalité IA : Fellah-GPT**  
Le chatbot Fellah-GPT permet d'interroger les documents suivants :  

📄 Lois agricoles marocaines  
🌦️ Zones agro-climatiques  
🌱 Fiches techniques de culture  

Accessible via un bouton flottant 💬 dans l'interface React (page d'accueil).  

---

### 🖼️ Aperçus visuels  
1. Interface du chatbot (RAG)  
2. Bouton flottant 💬 dans Home.jsx  
3. Backend RAG en exécution  

---

### 🤖 Technologies utilisées  
**Backend** : Flask, FAISS, LangChain, Google Generative AI  
**Frontend** : React, Vite, Lucide-react  
**Langages** : Python, JavaScript  

voici les captures d'écrans qui visualise ca  
![floating ](https://github.com/user-attachments/assets/c7e4cf6e-0827-4f63-bb54-fe42a9a77412)
![bienvenu message](https://github.com/user-attachments/assets/1895e4df-4e18-41f9-8dd8-0d3486822ad4)
![Capture d’écran 2025-06-04 010514](https://github.com/user-attachments/assets/83dc4dd9-b804-4ef5-a50e-937d0c71d163)


---

### 🔗 Projet d'origine  
[https://github.com/agri40/test-project-iot-watch.git](https://github.com/agri40/test-project-iot-watch.git)

achrafbouiguigne committed:
feat: ajout du chatbot RAG FellahGPT (backend + intégration React)

ismailaib reviewed (changes requested):
Salut @achrafbouiguigne , merci pour ta contribution 🙌
Quelques points à corriger pour rester alignés sur nos bonnes pratiques :

1. Le README ne doit pas servir de message de RP ou de changelog. Il doit documenter le projet tel qu’il existe actuellement (fonctionnalités, installation, usage, etc.). Merci de remettre l’ancien contenu ou de le compléter dans ce sens.

2. Les messages de commits ne respectent pas la convention attendue (feat:, fix:, chore:, etc.). C’est important pour garder un historique propre.

3. Pour faciliter la revue, pense à ajouter des captures d’écran ou des aperçus dans la description de la Pull Request. Ça m’aidera à mieux visualiser les changements attendus.

Merci pour ta compréhension 🙏

achrafbouiguigne commented:
Salut @ismailaib, merci pour ton retour ! Je vais remettre l’ancien contenu
du README et l’enrichir avec les infos nécessaires sur l’usage et
l’installation du projet. Je vais aussi corriger les messages de commits
pour suivre la convention feat: / fix: et ajouter des captures d’écran dans
la PR pour faciliter la revue. D’ailleurs, y a-t-il un délai spécifique
pour ce test ? Je souhaite ajouter une autre fonctionnalité avant de
finaliser. Merci encore

Le mar. 3 juin 2025 à 20:05, ismail Aitbouhmad ***@***.***> a
écrit :

> ***@***.**** requested changes on this pull request.
>
> Salut @achrafbouiguigne <https://github.com/achrafbouiguigne> , merci
> pour ta contribution 🙌
> Quelques points à corriger pour rester alignés sur nos bonnes pratiques :
>
>    1.
>
>    Le README ne doit pas servir de message de RP ou de changelog. Il doit
>    documenter le projet tel qu’il existe actuellement (fonctionnalités,
>    installation, usage, etc.). Merci de remettre l’ancien contenu ou de le
>    compléter dans ce sens.
>    2.
>
>    Les messages de commits ne respectent pas la convention attendue
>    (feat:, fix:, chore:, etc.). C’est important pour garder un historique
>    propre.
>    3.
>
>    Pour faciliter la revue, pense à ajouter des captures d’écran ou des
>    aperçus dans la description de la Pull Request. Ça m’aidera à mieux
>    visualiser les changements attendus.
>
> Merci pour ta compréhension 🙏
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Agri40-Stage/test-project-iot-watch/pull/52#pullrequestreview-2893742946>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHMG3BY53HZZRNCYSDZN4R33BXWWRAVCNFSM6AAAAAB6QE7NAGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQOJTG42DEOJUGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

achrafbouiguigne committed:
Update README.md

this is the modified README.md (the old on have serves PR)

achrafbouiguigne committed:
Merge branch 'dev' into feat/fellah-gpt

achrafbouiguigne committed:
Update README.md

achrafbouiguigne committed:
Update Header.jsx

this was aproblem form merging

achrafbouiguigne committed:
Update Header.jsx

achrafbouiguigne committed:
Update Header.jsx

achrafbouiguigne committed:
Update README.md

achrafbouiguigne commented:
J'espère bien que le conflit de merge est résolu maintenant
Le mar. 3 juin 2025 à 21:33, Achraf B ***@***.***> a écrit :

> Salut @ismailaib, merci pour ton retour ! Je vais remettre l’ancien
> contenu du README et l’enrichir avec les infos nécessaires sur l’usage et
> l’installation du projet. Je vais aussi corriger les messages de commits
> pour suivre la convention feat: / fix: et ajouter des captures d’écran
> dans la PR pour faciliter la revue. D’ailleurs, y a-t-il un délai
> spécifique pour ce test ? Je souhaite ajouter une autre fonctionnalité
> avant de finaliser. Merci encore
>
> Le mar. 3 juin 2025 à 20:05, ismail Aitbouhmad ***@***.***>
> a écrit :
>
>> ***@***.**** requested changes on this pull request.
>>
>> Salut @achrafbouiguigne <https://github.com/achrafbouiguigne> , merci
>> pour ta contribution 🙌
>> Quelques points à corriger pour rester alignés sur nos bonnes pratiques :
>>
>>    1.
>>
>>    Le README ne doit pas servir de message de RP ou de changelog. Il
>>    doit documenter le projet tel qu’il existe actuellement (fonctionnalités,
>>    installation, usage, etc.). Merci de remettre l’ancien contenu ou de le
>>    compléter dans ce sens.
>>    2.
>>
>>    Les messages de commits ne respectent pas la convention attendue
>>    (feat:, fix:, chore:, etc.). C’est important pour garder un historique
>>    propre.
>>    3.
>>
>>    Pour faciliter la revue, pense à ajouter des captures d’écran ou des
>>    aperçus dans la description de la Pull Request. Ça m’aidera à mieux
>>    visualiser les changements attendus.
>>
>> Merci pour ta compréhension 🙏
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Agri40-Stage/test-project-iot-watch/pull/52#pullrequestreview-2893742946>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BHMG3BY53HZZRNCYSDZN4R33BXWWRAVCNFSM6AAAAAB6QE7NAGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQOJTG42DEOJUGY>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***
>> com>
>>
>

achrafbouiguigne committed:
Update Header.jsx

i keep this to avoid merge conflicts (but the  header version in main that works very fine )

ismailaib reviewed (approved):
💯
